ingestion/src/main/java/feast/ingestion/transform/fn/ValidateFeatureRowDoFn.java

```diff
@@ -58,7 +58,9 @@ public static Builder newBuilder() {
   public void processElement(ProcessContext context) {
     String error = null;
     FeatureRow featureRow = context.element();
-    FeatureSet featureSet = getFeatureSets().getOrDefault(featureRow.getFeatureSet(), null);
+    featureRow =
+        featureRow.toBuilder().setFeatureSet(stripVersion(featureRow.getFeatureSet())).build();
+    FeatureSet featureSet = getFeatureSets().get(featureRow.getFeatureSet());
     List<FieldProto.Field> fields = new ArrayList<>();
     if (featureSet != null) {
       for (FieldProto.Field field : featureRow.getFieldsList()) {
```

Comment:
Should this for loop have `break` or `continue` if we are tolerant of unknown fields?

Reply:
Out of scope for this PR.

Reply:
Ok, but it seems like a problem because it is highlighted in the [Removal of Versions RFC](https://docs.google.com/document/d/1P44LHd724JloQtpn10naAg5MlzcrkeOlN3QzAn9O8uY/edit)?

 `Rows that are short of any fields, or have additional fields, will still be written to the stores`

Should we have an additional PR?

Reply:
oh right. I forgot. I'll add it to this pr, sorry!

Reply:
But the code already has that behavior?

Reply:
I'm just eyeballing it, so maybe I am missing something. Wouldn't the `break` statement stop the for loop entirely? Meaning if it finds an unknown field then it will stop processing further (possibly good) fields.

Reply:
ah crap sorry brain farted, you're right, it should be continue

Reply:
Also related to #628 

cc @junhui096